faunadb-scala/src/main/scala/faunadb/values/Codec.scala

```diff
@@ -158,4 +218,86 @@ trait Encoder[T] {
   def encode(t: T): Value
 }
 
+object Encoder {
+  implicit def ValueEncoder[T <: Value]: Encoder[T] = new Encoder[T] {
+    def encode(t: T): Value = if (t != null) t else NullV
+  }
+
+  implicit object BooleanEncoder extends Encoder[Boolean] {
+    def encode(t: Boolean) = BooleanV(t)
+  }
+
+  implicit object LocalDateEncoder extends Encoder[LocalDate] {
+    def encode(t: LocalDate) = if (t != null) DateV(t) else NullV
+  }
+
+  implicit object StringEncoder extends Encoder[String] {
+    def encode(t: String) = if (t != null) StringV(t) else NullV
+  }
+
+  implicit object CharEncoder extends Encoder[Char] {
+    def encode(t: Char) = LongV(t)
+  }
+
+  implicit object ByteEncoder extends Encoder[Byte] {
+    def encode(t: Byte) = LongV(t)
+  }
+
+  implicit object ShortEncoder extends Encoder[Short] {
+    def encode(t: Short) = LongV(t)
+  }
+
+  implicit object IntEncoder extends Encoder[Int] {
+    def encode(t: Int) = LongV(t)
+  }
+
+  implicit object LongEncoder extends Encoder[Long] {
+    def encode(t: Long) = LongV(t)
+  }
+
+  implicit object FloatEncoder extends Encoder[Float] {
+    def encode(t: Float) = DoubleV(t)
+  }
+
+  implicit object DoubleEncoder extends Encoder[Double] {
+    def encode(t: Double) = DoubleV(t)
+  }
+
+  trait ArrayEncoder[T] extends Encoder[Array[T]]
+
+  implicit def ArrayEncoder[T: Encoder](implicit ev: ClassTag[T]) = new ArrayEncoder[T] {
+    def encode(t: Array[T]): Value = if (t != null) ArrayV(t map { Value(_) } toVector) else NullV
+  }
+
+  implicit object ByteArrayEncoder extends ArrayEncoder[Byte] {
+    def encode(t: Array[Byte]): Value = if (t != null) BytesV(t) else NullV
+  }
+
+  implicit def CollectionEncoder[T: Decoder, Col[E] <: Traversable[E]](implicit ev: Encoder[T]): Encoder[Col[T]] = new Encoder[Col[T]] {
+    def encode(col: Col[T]) = if (col != null) ArrayV(col map { Value(_) } toVector) else NullV
+  }
+
+  implicit def OptionEncoder[T](implicit encoder: Encoder[T]): Encoder[Option[T]] = new Encoder[Option[T]] {
+    def encode(t: Option[T]) = t match {
+      case Some(v) => encoder.encode(v)
+      case None => NullV
+    }
+  }
+
+  implicit object NoneEncoder extends Encoder[None.type] {
```

Comment:
because `None` is an case object and not a class, `Some` will be encoded with `OptionEncoder`